src/function_spaces.jl

```diff
@@ -52,11 +52,27 @@ end
     length(dN) == n_base || throw(ArgumentError("dN must have length $(n_base)"))
 end
 
+#####################
+# Utility functions #
+#####################
+reference_volume{dim}(::FunctionSpace{dim, RefCube}) = 2^dim
+reference_volume{dim}(::FunctionSpace{dim, RefTetrahedron}) = 1 / factorial(dim)
+# For boundaries
+reference_volume(fs::FunctionSpace, ::Int) = reference_volume(fs_lower_dim(fs))
+reference_volume(fs::FunctionSpace{2, RefTetrahedron}, boundary::Int) = boundary == 1 ? sqrt(2) : 1.0
+reference_volume(fs::FunctionSpace{3, RefTetrahedron}, b::Int) = b == 3 ? sqrt(2 * 1.5) / 2.0 : 0.5
+n_boundaries{dim}(::FunctionSpace{dim, RefCube}) = 2*dim
+n_boundaries(::FunctionSpace{2, RefTetrahedron}) = 3
+n_boundaries(::FunctionSpace{3, RefTetrahedron}) = 4
+
 ############
 # Lagrange #
 ############
 type Lagrange{dim, shape, order} <: FunctionSpace{dim, shape, order} end
 
+fs_lower_dim{dim,shape,order}(::Lagrange{dim,shape,order}) = Lagrange{dim-1,shape,order}()
```

Comment:
I guess this is also quite bad naming, same as `fs_order`. But these are only used internally I think.